app/javascript/components/batch_update/components/BatchUpdateForm.spec.js

```diff
@@ -107,53 +158,118 @@ describe('with only null genes', () => {
+    component.setState(state)
+    wrapper.update()
+  }
 
-  it('renders the current genes', () => {
-    expect(wrapper.find(GeneInput).length).toEqual(2)
-    expect(wrapper.text()).toMatch('Kawaii')
-    expect(wrapper.text()).toMatch('Animals')
+  it('the "Queue" button is disabled prior to any pending changes', () => {
+    expect(wrapper.find('Button.queue button').prop('disabled')).toEqual(true)
   })
 
-  it('updates gene values from integers', () => {
-    component.onChangeGeneValue({ name: 'Kawaii', value: 70 })
-    expect(component.state['geneValues']).toMatchObject({
-      Kawaii: 70,
+  describe('changing genes', () => {
+    beforeEach(() => {
+      renderState({
+        geneValues: {
+          Kawaii: 0,
+          Animals: 100,
+        },
+      })
+    })
+
+    it('enables the "Queue" button', () => {
+      expect(wrapper.find('Button.queue button').prop('disabled')).toEqual(
+        false
+      )
+    })
+
+    it('renders the current genes', () => {
+      expect(wrapper.find(GeneInput).length).toEqual(2)
+      expect(wrapper.text()).toMatch('Kawaii')
+      expect(wrapper.text()).toMatch('Animals')
     })
-  })
 
-  it('updates gene values from number-ish strings', () => {
-    component.onChangeGeneValue({ name: 'Kawaii', value: '70' })
-    expect(component.state['geneValues']).toMatchObject({
-      Kawaii: 70,
+    it('updates gene values from integers', () => {
+      component.onChangeGeneValue({ name: 'Kawaii', value: 70 })
+      expect(component.state['geneValues']).toMatchObject({
+        Kawaii: 70,
+      })
+    })
+
+    it('updates gene values from number-ish strings', () => {
+      component.onChangeGeneValue({ name: 'Kawaii', value: '70' })
+      expect(component.state['geneValues']).toMatchObject({
+        Kawaii: 70,
+      })
+    })
+
+    it('nulls gene values from empty strings', () => {
+      component.onChangeGeneValue({ name: 'Kawaii', value: '' })
+      expect(component.state['geneValues']).toMatchObject({
+        Kawaii: null,
+      })
     })
   })
 
-  it('nulls gene values from empty strings', () => {
-    component.onChangeGeneValue({ name: 'Kawaii', value: '' })
-    expect(component.state['geneValues']).toMatchObject({
-      Kawaii: null,
+  describe('changing tags', () => {
+    beforeEach(() => {
+      renderState({
+        existingTags: defaultCommonTags,
+        tagsToAdd: ['bang'],
+      })
+    })
+
+    it('enables the "Queue" button', () => {
+      expect(wrapper.find('Button.queue button').prop('disabled')).toEqual(
+        false
+      )
+    })
+
+    it('renders the tag state', () => {
+      expect(wrapper.find(SelectedTag).length).toEqual(3)
+
+      expect(wrapper.text()).toMatch('foo')
+      expect(wrapper.text()).not.toMatch('-foo')
+      expect(wrapper.text()).not.toMatch('+foo')
+
+      expect(wrapper.text()).toMatch('bar')
+      expect(wrapper.text()).not.toMatch('-bar')
+      expect(wrapper.text()).not.toMatch('+bar')
```

Comment:
Ach, my bad for insisting on coverage for all this implementation-dependent stuff, since a lot of this will change with my next PR. But at least this makes it clear what I will need to change so 🙏